pkg/plugin/processor/plugin.go

```diff
@@ -0,0 +1,147 @@
+
+package processor
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+)
+
+// ProcessorPlugin receives records, manipulates them and returns back the processed
+// records.
+//
+//nolint:revive // This interface will be moved into the SDK
+type ProcessorPlugin interface {
+	// Specification contains the metadata of this processor like name, version,
+	// description and a list of parameters expected in the configuration.
+	Specification() Specification
+
+	// Configure is the first function to be called in a processor. It provides the
+	// processor with the configuration that needs to be validated and stored.
+	// In case the configuration is not valid it should return an error.
+	// Configure should not open connections or any other resources. It should solely
+	// focus on parsing and validating the configuration itself.
+	Configure(context.Context, map[string]string) error
+
+	// Open is called after Configure to signal the processor it can prepare to
+	// start writing records. If needed, the processor should open connections and
+	// start background jobs in this function.
+	Open(context.Context) error
+
+	// Process takes a number of records and processes them right away.
+	// It should return a slice of ProcessedRecord that matches the length of
+	// the input slice. If an error occurred while processing a specific record
+	// it should be reflected in the ProcessedRecord with the same index as the
+	// input record that caused the error.
+	// Process should be idempotent, as it may be called multiple times with the
+	// same records (e.g. after a restart when records were not flushed).
+	Process(context.Context, []opencdc.Record) []ProcessedRecord
+
+	// Teardown signals to the processor that the pipeline is shutting down and
+	// there will be no more calls to any other function. After Teardown returns,
+	// the processor will be discarded.
+	Teardown(context.Context) error
+}
+
+// Specification is returned by a plugin when Specify is called.
+// It contains information about the configuration parameters for plugins
+// and allows them to describe their parameters.
+type Specification struct {
+	// Name is the name of the plugin.
+	Name string
+	// Summary is a brief description of the plugin and what it does.
+	Summary string
+	// Description is a more long form area appropriate for README-like text
+	// that the author can provide for documentation about the specified
+	// Parameters.
+	Description string
+	// Version string. Should be a semver prepended with `v`, e.g. `v1.54.3`.
+	Version string
+	// Author declares the entity that created or maintains this plugin.
+	Author string
+	// SourceParams and DestinationParams are maps of named Parameters that
+	// describe how to configure the plugins Destination or Source.
+	SourceParams      map[string]Parameter
+	DestinationParams map[string]Parameter
```

Comment:
This comes probably from the connector SDK, but I believe we need only params for processors.